share/wake/lib/system/http.wake

```diff
@@ -0,0 +1,199 @@
+package http
+
+from http export type HttpRequest
+
+# The http methods that may be called against a web server.
+export data HttpMethod =
+    HttpMethodGet
+    HttpMethodHead
+    HttpMethodPost
+    HttpMethodPut
+    HttpMethodDelete
+    HttpMethodConnect
+    HttpMethodOptions
+    HttpMethodTrace
+    HttpMethodPatch
+
+# The headers that may be assocated with an http request.
+#
+# These are often used to specify how to interpret the body or to provide authorization.
+tuple HttpHeader =
+    Name: String
+    Value: String
+
+# The description of a http request to send to a web server.
+tuple HttpRequest =
+    # The url of the function to be called
+    Url: String
+    # The method to be called on the destination
+    Method: HttpMethod
+    # The headers to be set on the request
+    Headers: List HttpHeader
+    # The body of the request
+    Body: String
+
+export tuple HttpResponse =
+    # The status code returned from the web server
+    StatusCode: Integer
+    # The headers returned from the web server
+    Headers: List HttpHeader
+    # The body returned from the web server
+    Body: String
```

Comment:
Since you're exporting the constructor/destructor, I expect this is just the standard confusion over tuple syntax:
```suggestion
export tuple HttpResponse =
    # The status code returned from the web server
    export StatusCode: Integer
    # The headers returned from the web server
    export Headers: List HttpHeader
    # The body returned from the web server
    export Body: String
```

Reply:
yep, had already fixed that locally, thanks